code/model/TimedNotice.php

```diff
@@ -178,6 +184,60 @@ public function canCreate($member = null)
     }
 
     /**
+     * Gets any notices relevant to the present time, context and current users
+     *
+     * @return ArrayList
+     **/
+    public static function getNotices()
+    {
+        // analyse the context this message gets requested
+        $context = self::getContext();
+
+        // prepare and filter the possible result
+        $now        = SS_Datetime::now()->getValue();
+        $member     = Member::currentUser();
+        $notices    = TimedNotice::get()->where("
+            Context = '{$context}' AND
+            StartTime < '{$now}' AND
+            (EndTime > '{$now}' OR EndTime IS NULL)
+        ");
+
+        // if there are notices verify if those are allowed for this group
+        if ($notices->count()) {
+            // turn the DataList into an ArrayList to make it editable.
+            $notices = ArrayList::create($notices->toArray());
+
+            foreach ($notices as $notice) {
+                if ($notice->CanViewType == 'OnlyTheseUsers') {
+                    if ($member && !$member->inGroups($notice->ViewerGroups())) {
+                        $notices->remove($notice);
+                    }
+                }
+            }
+        }
+
+        return $notices;
+    }
+
+    /**
+     * Analysis the context and returns the context information as string
+     *
+     * @return string
+     */
+    public static function getContext()
```

Comment:
This method causes an issue when editing a TimedNotice in the CMS. If you select "CMS" as the context, then save, the dropdown will be reloaded with "Website" as the selected value. We'll possibly need to rename this method to something other than getContext, so that it doesn't intercept attempts to get the value from the DB.